### PR TITLE
Bluetooth: Controller: df: Relax cl CTE RX enable param check 

### PR DESCRIPTION
**Pull request by ppryga-nordic**
According to Core 5.3 Vol 4, Part E, section 7.8.82 slot_durations,
switch_pattern_len and ant_ids are used only for AoA and do not affect
reception of AoD CTE.

To improve interoperability checking of correctness of these parameters
is not required. It will not affect lower link layer in case the AoA
CTE reception feature is not enabled in Kconfig.

BTW. Added a commit that fixes some spelling errors in `ull_df.c`.

Signed-off-by: Piotr Pryga <piotr.pryga@nordicsemi.no>

### CODE REVIEW

File: subsys/bluetooth/controller/ll_sw/ull_df.c [hunk 1]
```diff
@@ -456,35 +456,41 @@ uint8_t ll_df_set_cl_iq_sampling_enable(uint16_t handle,
...
 
 		/* Enable of already enabled CTE updates AoA configuration */
-		if (!((IS_ENABLED(CONFIG_BT_CTLR_DF_ANT_SWITCH_1US) &&
-		       slot_durations == BT_HCI_LE_ANTENNA_SWITCHING_SLOT_1US) ||
-		      slot_durations == BT_HCI_LE_ANTENNA_SWITCHING_SLOT_2US)) {
-			return BT_HCI_ERR_UNSUPP_FEATURE_PARAM_VAL;
+
+		/* According to Core 5.3 Vol 4, Part E, section 7.8.82 slot_durations,
+		 * switch_pattern_len and ant_ids are used only for AoA and do not affect
+		 * reception of AoD CTE. If AoA is not supporeted relax command validation
```
Comment (kruithofa) [inline subsys/bluetooth/controller/ll_sw/ull_df.c:462]:
```suggestion
		 * reception of AoD CTE. If AoA is not supported relax command validation
```

File: subsys/bluetooth/controller/ll_sw/ull_df.c [hunk 2]
```diff
@@ -1210,6 +1216,11 @@ uint8_t ll_df_set_conn_cte_rx_params(uint16_t handle, uint8_t sampling_enable,
 	if (!sampling_enable) {
 		params_rx->is_enabled = false;
 	} else {
+		/* According to Core 5.3 Vol 4, Part E, section 7.8.83 slot_durations,
+		 * switch_pattern_len and ant_ids are used only for AoA and do not affect
+		 * reception of AoD CTE. If AoA is not supporeted relax command validation
```
Comment (kruithofa) [inline subsys/bluetooth/controller/ll_sw/ull_df.c:1221]:
```suggestion
		 * reception of AoD CTE. If AoA is not supported relax command validation
```

File: subsys/bluetooth/controller/ll_sw/ull_df.c (hunk 1 above)
Comment (ppryga-nordic) [reply subsys/bluetooth/controller/ll_sw/ull_df.c]: Corrected

File: subsys/bluetooth/controller/ll_sw/ull_df.c (hunk 2 above)
Comment (ppryga-nordic) [reply subsys/bluetooth/controller/ll_sw/ull_df.c]: Corrected

File: subsys/bluetooth/controller/ll_sw/ull_df.c [hunk 3]
```diff
@@ -402,7 +402,7 @@ uint8_t ll_df_set_cl_cte_tx_enable(uint8_t adv_handle, uint8_t cte_enable)
  * @return Status of command completion.
  *
  * @Note This function may put TX thread into wait state. This may lead to a
- *       situation that ll_sync_set instnace is relased (RX thread has higher
+ *       situation that ll_sync_set instance is relased (RX thread has higher
  *       priority than TX thread). l_sync_set instance may not be accessed after
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ull_df.c:406]:
```suggestion
 *       priority than TX thread). ll_sync_set instance may not be accessed after
```

File: subsys/bluetooth/controller/ll_sw/ull_df.c [hunk 4]
```diff
@@ -456,35 +456,41 @@ uint8_t ll_df_set_cl_iq_sampling_enable(uint16_t handle,
...
+			       slot_durations == BT_HCI_LE_ANTENNA_SWITCHING_SLOT_1US) ||
+			      slot_durations == BT_HCI_LE_ANTENNA_SWITCHING_SLOT_2US)) {
+				return BT_HCI_ERR_UNSUPP_FEATURE_PARAM_VAL;
+			}
+
+			if (switch_pattern_len < BT_HCI_LE_SWITCH_PATTERN_LEN_MIN ||
+			    switch_pattern_len > BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN || !ant_ids) {
+				return BT_HCI_ERR_UNSUPP_FEATURE_PARAM_VAL;
+			}
+			memcpy(cfg->ant_ids, ant_ids, switch_pattern_len);
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ull_df.c:476]: Blank line after if-clause, and `(void)` for `memcpy`.

```suggestion

			(void)memcpy(cfg->ant_ids, ant_ids, switch_pattern_len);
```

**Review by kruithofa — DISMISSED** [review]
LGTM

**Review by cvinayak — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/controller/ll_sw/ull_df.c (hunk 4 above)
Comment (ppryga-nordic) [reply subsys/bluetooth/controller/ll_sw/ull_df.c]: Corrected

**Review by kruithofa — APPROVED** [review]

**Review by cvinayak — APPROVED** [review]

**Pull request merged**